examples/flower_classifier/keras_image_classifier.py

```diff
@@ -0,0 +1,364 @@
+        :param optimizer: Keras optimizer to be used to train the model.
+        :param seed: Random seed. Used e.g. when spliting the dataset into train / validation.
+        """
+        assert len(set(labels)) == len(domain)
+        with mlflow.start_run() as run:
+            for param, value in self.params().items():
+                mlflow.log_param(str(param), str(value))
+
+            mlflow.log_param("epochs", str(epochs))
+            mlflow.log_param("batch_size", str(batch_size))
+            mlflow.log_param("validation_ratio", str(test_ratio))
+            mlflow.log_param("optimizer", optimizer.__name__)
+            if seed:
+                mlflow.log_param("seed", str(seed))
+
+            with tf.Graph().as_default() as graph:
+                import keras.backend as K
+                K.set_session(tf.Session(graph=graph))
+                dims = self._input_shape[:2]
+                x = np.array([decode_and_resize_image(read_image(x), dims) for x in image_files])
+                y = np_utils.to_categorical(np.array(labels), num_classes=len(domain))
+                x_train, x_valid, y_train, y_valid = train_test_split(x, y, random_state=seed,
+                                                                      train_size=(1 - test_ratio))
+                model = self.create_model(classes=len(domain))
+                model.compile(
+                    optimizer=optimizer(),
+                    loss=tf.keras.losses.categorical_crossentropy,
+                    metrics=["accuracy"])
+                model.fit(
+                    x=x_train,
+                    y=y_train,
+                    validation_data=(x_valid, y_valid),
+                    epochs=epochs,
+                    batch_size=batch_size,
+                    callbacks=[MLflowLogger(model=model,
+                                            x_train=x_train,
+                                            y_train=y_train,
+                                            x_valid=x_valid,
+                                            y_valid=y_valid,
+                                            artifact_path="model",
+                                            domain=sorted(domain.keys(), key=lambda x: domain[x]),
+                                            image_dims=self._input_shape)])
+
+
+class KerasImageClassifierPyfunc(object):
+    """
+    PyFunc for Keras Image Classifier.
+
+    Input is expected to be raw base64 encoded images. The images are decoded and resized prior to
+    passing them to the model.
+
+    The prediction output is the predicted class id.
+    """
+
+    def __init__(self, model, image_dims, domain):
+        self._model = model
+        self._image_dims = image_dims
+        self._domain = domain
+        probs_names = ["p({})".format(x) for x in domain]
+        self._column_names = ["predicted_label_id", "predicted_label"] + probs_names
+
+    def predict_images(self, images):
```

Comment:
Do we want the user to call this method directly? If so, we should document it. If not, we should make it "internal" by prepending an underscore to the method name.